utils/maybe_async/README.md

```diff
@@ -1,91 +1,47 @@
 # Winter maybe-async
-This crate contains a `maybe_async` proc macro that abstracts away sync/async. It is heavily based on [`maybe-async`](https://github.com/fMeow/maybe-async-rs).
 
-The `maybe_async` macro will generate a synchronous or asynchronous version of the trait it is marking. To generate the asynchronous version, enable the `async` feature on the crate. If the `async` feature is off, the synchronous version will be generated. For example,
+This crate contains the `maybe_async` procedural attribute macro and the `maybe_await` procedural macro which abstract away Rust sync/async.
 
-```rs
-#[maybe_async]
-trait ExampleTrait {
-    async fn say_hello(&self) {
-        let hello = self.get_hello().await;
+## maybe_async
 
-        println!("{}", hello);
-    }
+The `maybe_async` macro will conditionally add the `async` keyword to a function it marks depending on the `async` feature being enabled. To generate the asynchronous version, enable the `async` feature on the crate. If the `async` feature is off, the synchronous version will be generated. For example,
 
-    async fn get_hello(&self) -> String {
-        "hello".into()
-    }
-}
-
-// Generate code when `async` feature is turned ON
-#[async_trait]
+```rust
+// Adding `maybe_async` to trait functions
 trait ExampleTrait {
-    async fn say_hello(&self) {
-        let hello = self.get_hello().await;
-
-        println!("{}", hello);
-    }
+    #[maybe_async]
+    fn say_hello(&self);
 
-    async fn get_hello(&self) -> String {
-        "hello".into()
-    }
+    #[maybe_async]
+    fn get_hello(&self) -> String;
 }
 
-// Generate code when `async` feature is turned OFF
-trait ExampleTrait {
-    fn say_hello(&self) {
-        let hello = self.get_hello();
-
-        println!("{}", hello);
-    }
-
-    fn get_hello(&self) -> String {
-        "hello".into()
-    }
+// Adding `maybe_async` to regular functions
+#[maybe_async]
+fn hello_world() {
+    // ...
 }
 ```
```

Comment:
I would probably add how the output would look like if `async` feature is on. Something like:

When the `async` feature is enabled, the above code will be transformed into:
```Rust
async trait ExampleTrait {
    async fn say_hello(&self);

    async fn get_hello(&self) -> String;
}

async fn hello_world() {
    // ...
}
```